enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -178,15 +178,19 @@ Eg. if Dashboard B includes a Card A that is derived from a
     (escape-report analysis)
     ;; If it's nil, there are no errors, and we can proceed to do the dump.
     (let [closure  (descendants-closure targets)
+          models   (model-set opts)
           ;; filter the selected models based on user options
           by-model (-> (group-by first closure)
-                       (select-keys (model-set opts))
+                       (select-keys models)
                        (update-vals #(set (map second %))))]
-      (eduction (map (fn [[model ids]]
-                       (eduction (map #(serdes/extract-one model opts %))
-                                 (db/select-reducible (symbol model) :id [:in ids]))))
-                cat
-                by-model))))
+      (concat
```

Comment:
This seems overly complicated, and hopefully `concat`ing two `eduction`s together is OK, since `concat` produces a lazy seq, it should not interfere with the reducible query bit.

Reply:
An important difference is lazy seqs are cached whereas eductions are not. That is, reduction an eduction twice would call the operations in it twice. With a lazy sequence the operations would only be called once (the first time).